models/ios/vlans/ios_vlans.yaml

```diff
@@ -0,0 +1,56 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: ios
+RESOURCE: vlans
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: ios_vlans
+  version_added: 2.9
+  short_description: Manage VLANs on Cisco IOS devices.
+  description: This module provides declarative management of VLANs on Cisco IOS network devices.
+  author: Sumit Jaiswal (@justjais)
+  notes:
+  - Tested against Cisco IOSv Version 15.2 on VIRL
+  options:
+    config:
+      description: A dictionary of VLANs options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Ascii name of the VLAN.
+          type: str
+        vlan_id:
+          description:
+          - ID of the VLAN. Range 1-4094
+          type: int
+          required: True
+        interfaces:
+          description:
+          - List of interfaces that should be associated to the VLAN.
```

Comment:
@cidrblock do we need interface assignment to be part of vlan RM or this will be handled in vlan_interfaces?  tagging @justjais just FYI.

Reply:
No need for interfaces here, the per interface vlan configuration work is handles in the l2_interfaces module

Reply:
yea makes sense, I'll update the model.